lib/node_modules/@stdlib/array/float16/lib/polyfill.js

```diff
@@ -0,0 +1,903 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var min = require( '@stdlib/math/base/special/min' );
+var max = require( '@stdlib/math/base/special/max' );
+var floor = require( '@stdlib/math/base/special/floor' );
+
+
+// MAIN //
+
+/**
+* Creates a new Float16Array.
+*
+* @constructor
+* @param {(number|Array|ArrayBuffer|TypedArray)} input - Input data or buffer
+* @param {NonNegativeInteger} [byteOffset=0] - Byte offset
+* @param {NonNegativeInteger} [length] - Number of elements
+* @throws {TypeError} Must provide valid input
+*/
+function Float16Array( input, byteOffset, length ) {
+    var i;
+    var len;
+    var offset;
+
+    byteOffset = byteOffset || 0;
+
+    if ( typeof input === 'number' ) {
+        this._buffer = new Uint16Array( input );
+    } else if ( input instanceof ArrayBuffer ) {
+        offset = byteOffset >>> 0;
+        len = ( typeof length !== 'undefined' ) ? ( length >>> 0 ) : ( ( input.byteLength - offset) / 2 );
+        this._buffer = new Uint16Array( input, offset, len );
+    } else if (Object.prototype.toString.call( input ) === '[object Array]' ) {
+        this._buffer = new Uint16Array( input.length );
+        for (i = 0; i < input.length; i++) {
+            this._buffer[ i ] = Float16Array.toFloat16( input[ i ] );
+        }
+    } else if ( typeof input === 'object' && typeof input.length === 'number' ) {
+        this._buffer = new Uint16Array( input.length );
+        for ( i = 0; i < input.length; i++ ) {
+            this._buffer[ i ] = Float16Array.toFloat16( input[ i ] );
+        }
+    } else {
+        throw new TypeError( 'Invalid argument. Must provide a number, Array, ArrayBuffer, or typed array. Value: ' + input + '.' );
+    }
+
+    this.length = this._buffer.length;
+}
+
+/**
+* Get the underlying `ArrayBuffer`.
+*
+* @name buffer
+* @memberof Float16Array.prototype
+* @readonly
+* @type {ArrayBuffer}
+*/
+Object.defineProperty( Float16Array.prototype, 'buffer', {
```

Comment:
As in `array/bool`, you should be using our utilities for defining properties. E.g., `utils/define-nonenumerable-read-only-accessor`. Right now, `buffer` is enumerable, which it shouldn't be.